jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -205,15 +205,32 @@ def _on_display_units_changed(self, event={}):
                 # convert background to new unit
                 if self.background_value is not None:
 
-                    prev_unit = u.Unit(prev_display_unit)
-                    new_unit = u.Unit(self.display_unit)
+                    # FIXME: Kinda hacky, better solution welcomed.
+                    # Basically we want to forget about PIX2 and do normal conversion.
+                    # Since traitlet does not keep unit, we do not need to reapply PIX2.
+                    if "pix2" in prev_display_unit and "pix2" in self.display_unit:

```

Comment:
can you check if this is the case for steradian as well? i believe you cant convert flux/sr <> flux/sr if there is a u.spectral_density equivalency needed.

you could also leave a note here to pass this through utils.flux_conversion - i'm working on generalizing this and it might simplify this code if we want to swap it later on

Reply:
I think steradian is already covered by existing equivalencies combo because steradian is a recognized physical unit in astronomy, whereas pixel is very much tied to instrumentation and not well defined in `astropy.units`. CI passes without such treatment for `u.sr`. Unless you mean somehow the existing test suite does not cover the steradian case already?

https://github.com/spacetelescope/jdaviz/blob/fc88465b8feb8c5bbc7dbee7e955610ada1ba732/jdaviz/utils.py#L519

Reply:
Ah, I see the test I unskipped is about Cubeviz unit conversion plugin only. And this code is inside aperture photometry. I can think of how I can expand your test case to go through this line as well tomorrow, maybe. 🤔

Reply:
The case i was thinking of where you'd need another workaround is Jy <> erg/s/cm2/A/sr, which won't convert with the spectral density equivalency alone for some reason and you need to also remove the angle component. I dont think that will be encountered here though since it will always be surface brightness to surface brightness conversions

Reply:
I don't think we need to worry about conversions that will never happen in aperture photometry background calculation? Only the aperture sum removes a solid angle. Background is a simple median and should keep data unit because you subtract background before you sum it.

I also added background calculation test in the test you wrote that I unskipped here. Hopefully that will put your mind at ease?